process/connections.go

```diff
@@ -34,6 +34,26 @@ func (m *CollectorConnections) IterateDNS(addr *Addr, cb func(i, total int, entr
 }
 
 // GetDNSNames returns all the DNS entries
-func (m *CollectorConnections) GetDNSNames() []string {
+func (m *CollectorConnections) GetDNSNames() ([]string, error) {
 	return getDNSNames(m.EncodedDNS)
 }
+
+// GetDNSV2 returns the DNS entries for the given addr.
+// The first argument returned is the first DNS entry followed by any additional name resolutions.  Most IPs will
+// have a single resolution so this dual format allows us to avoid allocations for the common case.  If there are
+// multiple name resolutions, there is no implied priority between the dual values
+// the return values are indexes into the dns database (GetDNSDatabase)
+
+func (m *CollectorConnections) GetDNSV2(addr *Addr) (int32, []int32, error) {
```

Comment:
Can we keep the same interface for V2?
In on other words, instead of returning indexes, can we return the domain strings directly as we do for `V1`? Same comment for the function `IterateDNSV2`.

Reply:
We could, but that I think would defeat the entire thing.  We'd then decompose the list every time.  Especially for `Iterate`, which is used even when you're looking for a single string.

Reply:
Since `GetDNS` now takes care of calling the right version, this is no longer needed.